odoo/fields.py

```diff
@@ -4246,7 +4246,13 @@ def convert_to_record(self, value, record):
             Comodel._active_name
             and self.context.get('active_test', record.env.context.get('active_test', True))
         ):
-            corecords = corecords.filtered(Comodel._active_name).with_prefetch(prefetch_ids)
+            try:
+                corecords = corecords.filtered(Comodel._active_name)
+            except AccessError:
```

Comment:
You should filter accessible records only or just raise the exception.

Reply:
No the whole point is to not do that...